Description: Fix crash when gathering system information on newer Gnome
 versions.
 There is no <date> tag in recent versions of the gnome-version.xml so the variable
 gnomeB is never set to true which causes the loop to run until the end of the file
 is reached and there is no check if textread.ReadLine() returns null before any
 string operations are performed on the result.
Author: Alexander Grothe <alexander.grothe@googlemail.com>
Origin: other, https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=816481;msg=10
Bug-Debian: https://bugs.debian.org/816481
Bug-Ubuntu: https://launchpad.net/bugs/1575182
Forwarded: No, project seems dead upstream
Last-Update: 2017-05-24

--- sysinfo-0.7.orig/Sysinfo/SystemInfo.cs
+++ sysinfo-0.7/Sysinfo/SystemInfo.cs
@@ -171,9 +171,7 @@ namespace Sysinfo {

                                using (TextReader textread = File.OpenText(gnome_about)) {

-                                       while ( gnomeB == false ) {
-
-                                               temp = textread.ReadLine();
+                                       while ( (temp = textread.ReadLine()) != null && gnomeB == false ) {

                                                //get version from xml
                                                if ( temp.EndsWith("platform>")) {
